onnxruntime/python/tools/transformers/longformer/convert_longformer_to_onnx.py

```diff
@@ -80,134 +78,67 @@ def parse_arguments():
-                                                            self.one_sided_attn_window_size)
-
-    assert attn_output.size() == hidden_states.size(), "Unexpected size"
-
-    outputs = (attn_output, )
-    return outputs
-
-
-#For transformers 4.0
+# For transformers 4.0.0
 def my_longformer_self_attention_forward_4(self,
                                            hidden_states,
                                            attention_mask=None,
                                            is_index_masked=None,
                                            is_index_global_attn=None,
                                            is_global_attn=None):
-    # TODO: move mask calculation to LongFormerModel class to avoid calculating it again and again in each layer.
     global_mask = is_index_global_attn.int()
-    torch.masked_fill(attention_mask, is_index_global_attn, 0.0)
+    # The following check is based on the dummy inputs (only the first token is global).
+    assert len(global_mask.shape) == 2 and global_mask.shape[0] == 1 and global_mask.count_nonzero().item(
+    ) == 1 and global_mask.tolist()[0][0] == 1
 
-    weight = torch.stack(
-        (self.query.weight.transpose(0, 1), self.key.weight.transpose(0, 1), self.value.weight.transpose(0, 1)), dim=1)
-    weight = weight.reshape(self.embed_dim, 3 * self.embed_dim)
-
-    bias = torch.stack((self.query.bias, self.key.bias, self.value.bias), dim=0)
-    bias = bias.reshape(3 * self.embed_dim)
-
-    global_weight = torch.stack((self.query_global.weight.transpose(0, 1), self.key_global.weight.transpose(
-        0, 1), self.value_global.weight.transpose(0, 1)),
-                                dim=1)
-    global_weight = global_weight.reshape(self.embed_dim, 3 * self.embed_dim)
-
-    global_bias = torch.stack((self.query_global.bias, self.key_global.bias, self.value_global.bias), dim=0)
-    global_bias = global_bias.reshape(3 * self.embed_dim)
+    input_mask = is_index_masked.float()
+    input_mask = input_mask.masked_fill(is_index_masked, -10000.0)
+    # Yet another way to generate input_mask = torch.masked_fill(attention_mask, is_index_global_attn, 0.0)
 
-    attn_output = torch.ops.onnxruntime.LongformerAttention(hidden_states, weight, bias, attention_mask, global_weight,
-                                                            global_bias, global_mask, self.num_heads,
-                                                            self.one_sided_attn_window_size)
-
-    assert attn_output.size() == hidden_states.size(), "Unexpected size"
+    # TODO: add postprocess of ONNX model to calculate based on graph input: input_mask = (attention_mask - 1) * 10000.0
+    # TODO: add postprocess of ONNX model to use graph input directly: glboal_mask = global_attention_mask
 
-    outputs = (attn_output, )
-    return outputs
-
-
-# For transformers 3.0
-def my_longformer_attention_forward_3(self, hidden_states, attention_mask, output_attentions=False):
```

Comment:
Will we support transformers 3.0?

Reply:
I removed the support for transformers 3.0 since the ONNX graph for 3.0 is slower. In this way, we could push user to use 4.0 for better performance, and reduce maintenance cost.